app/src/main/java/com/zeapo/pwdstore/git/GitOperation.kt

```diff
@@ -175,70 +176,113 @@ abstract class GitOperation(fileDir: File, internal val callingActivity: Activit
                                     .setMessage(callingActivity.resources.getString(R.string.passphrase_dialog_text))
                                     .setView(dialogView)
                                     .setPositiveButton(callingActivity.resources.getString(R.string.dialog_ok)) { _, _ ->
                                         if (keyPair.decrypt(passphrase.text.toString())) {
-                                            val rememberPassphrase = (dialogView.findViewById<View>(R.id.sshkey_remember_passphrase) as CheckBox).isChecked
+                                            val rememberPassphrase = dialogView.findViewById<MaterialCheckBox>(R.id.git_auth_remember_passphrase).isChecked
                                             if (rememberPassphrase) {
-                                                settings.edit().putString("ssh_key_passphrase", passphrase.text.toString()).apply()
+                                                encryptedSettings.edit().putString("ssh_key_passphrase", passphrase.text.toString()).apply()
                                             }
                                             // Authenticate using the ssh-key and then execute the command
                                             setAuthentication(sshKey, username, passphrase.text.toString()).execute()
                                         } else {
-                                            settings.edit().putString("ssh_key_passphrase", null).apply()
+                                            encryptedSettings.edit().putString("ssh_key_passphrase", null).apply()
                                             // call back the method
                                             executeAfterAuthentication(connectionMode, username, sshKey, identity, true)
                                         }
-                                    }.setNegativeButton(callingActivity.resources.getString(R.string.dialog_cancel)) { _, _ ->
-                                        // Do nothing.
-                                    }.show()
+                                    }
+                                    .setNegativeButton(callingActivity.resources.getString(R.string.dialog_cancel), null)
+                                    .create()
+                            dialog.setOnShowListener {
+                                passphrase.apply {
+                                    setOnFocusChangeListener { v, _ ->
+                                        v.post {
+                                            val imm = callingActivity.getSystemService<InputMethodManager>()
+                                            imm?.showSoftInput(v, InputMethodManager.SHOW_IMPLICIT)
+                                        }
+                                    }
+                                    requestFocus()
+                                }
+                            }
+                            dialog.show()
```

Comment:
Dismissing this dialog with a back press does not automatically dismiss the underlying activity. Maybe add a dismiss/cancel listener here?

Reply:
Fixed with 1e94e5910de5